Sources/PushNotifications.swift

```diff
@@ -8,26 +8,43 @@ import NotificationCenter
 import Foundation
 
 @objc public final class PushNotifications: NSObject {
-    private var deviceId: String?
-    private var instanceId: String?
-    private var baseURL: String?
     private let session = URLSession.shared
     //! Returns a shared singleton PushNotifications object.
     /// - Tag: shared
     @objc public static let shared = PushNotifications()
 
     /**
-     Register with PushNotifications service.
+     Start PushNotifications service.
 
      - Parameter instanceId: PushNotifications instance id.
 
      - Precondition: `instanceId` should not be nil.
      */
-    @objc public func register(instanceId: String) {
-        self.instanceId = instanceId
-        self.baseURL = "https://\(instanceId).pushnotifications.pusher.com/device_api/v1/instances"
-        self.registerForPushNotifications()
+    @objc public func start(instanceId: String) {
+        do {
+            try Instance.persist(instanceId)
+        } catch PusherAlreadyRegisteredError.instanceId(let errorMessage) {
+            print(errorMessage)
+        } catch {
+            print("Unexpected error: \(error).")
+        }
+    }
+
+    /**
+     Register to receive remote notifications via Apple Push Notification service.
+     */
+    @objc public func registerForRemoteNotifications() {
```

Comment:
Don't we want to pass the options as a parameter that defaults to `[.alert, .sound, .badge]`?